mcs/class/Makefile

```diff
@@ -105,6 +105,8 @@ monodroid_dirs := \
 	$(mobile_common_dirs)	\
 	Mono.CompilerServices.SymbolWriter	\
 	Mono.Btls.Interface	\
+	System.Runtime.Serialization.Formatters.Soap	\
+	System.Runtime.Remoting	\
```

Comment:
Without remoting, only the most primitive forms of cross-AppDomain communications work. It's required for transparent proxies and MarshalByRef objects.

We explicitly don't support AppDomains or Remoting on iOS.
Android was left as a maybe and we end up deciding to support AppDomains.